app/controllers/nonsipp/loansmadeoroutstanding/DatePeriodLoanController.scala

```diff
@@ -0,0 +1,143 @@
+import services.{SaveService, SchemeDateService}
+import utils.DateTimeUtils.localDateShow
+import viewmodels.DisplayMessage.Message
+import viewmodels.implicits._
+import viewmodels.models.MultipleQuestionsViewModel.TripleQuestion
+import viewmodels.models.{FormPageViewModel, QuestionField}
+import views.html.MultipleQuestionView
+import controllers.nonsipp.loansmadeoroutstanding.DatePeriodLoanController._
+
+import java.time.LocalDate
+import javax.inject.{Inject, Named}
+import scala.concurrent.{ExecutionContext, Future}
+
+class DatePeriodLoanController @Inject()(
+  override val messagesApi: MessagesApi,
+  saveService: SaveService,
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  val controllerComponents: MessagesControllerComponents,
+  view: MultipleQuestionView,
+  schemeDateService: SchemeDateService
+)(implicit ec: ExecutionContext)
+    extends PSRController {
+
+  def onPageLoad(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) { implicit request =>
+    schemeDateService.taxYearOrAccountingPeriods(srn).merge.getOrRecoverJourney { date =>
+      Ok(
+        view(
+          viewModel(
+            srn,
+            request.schemeDetails.schemeName,
+            mode,
+            request.userAnswers.fillForm(DatePeriodLoanPage(srn), form(date.to))
+          )
+        )
+      )
+    }
+  }
+
+  def onSubmit(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn).async { implicit request =>
+    schemeDateService.taxYearOrAccountingPeriods(srn).merge.getOrRecoverJourney { date =>
+      form(date.to)
+        .bindFromRequest()
+        .fold(
+          formWithErrors =>
+            Future.successful {
+              BadRequest(view(viewModel(srn, request.schemeDetails.schemeName, mode, formWithErrors)))
+            },
+          value =>
+            for {
+              updatedAnswers <- Future.fromTry(request.userAnswers.transformAndSet(DatePeriodLoanPage(srn), value))
+              _ <- saveService.save(updatedAnswers)
+            } yield Redirect(navigator.nextPage(DatePeriodLoanPage(srn), mode, updatedAnswers))
+        )
+    }
+  }
+}
+
+object DatePeriodLoanController {
+
+  private def field1Errors(date: LocalDate)(implicit messages: Messages): DateFormErrors =
+    DateFormErrors(
+      "datePeriodLoan.field1.error.required.all",
+      "datePeriodLoan.field1.error.required.day",
+      "datePeriodLoan.field1.error.required.month",
+      "datePeriodLoan.field1.error.required.year",
+      "datePeriodLoan.field1.error.required.two",
+      "datePeriodLoan.field1.error.invalid.date",
+      "datePeriodLoan.field1.error.invalid.characters",
+      List(DateFormErrors.failIfDateAfter(date, messages("datePeriodLoan.field1.error.future", date.show)))
+    )
+
+  private val field2Errors: MoneyFormErrors =
+    MoneyFormErrors(
+      "datePeriodLoan.field2.error.required",
```

Comment:
Value and period of loan can be negative, is that right?

Reply:
Yeah thats right 👍